apps/aechannel/src/aesc_client_reconnect_tx.erl

```diff
@@ -0,0 +1,185 @@
+-module(aesc_client_reconnect_tx).
+
+-behavior(aetx).
+-behavior(aesc_signable_transaction).
+
+%% Behavior API
+-export([
+          new/1
+        , type/0
+        , fee/1
+        , gas/1
+        , ttl/1
+        , nonce/1
+        , origin/1
+        , check/3
+        , process/3
+        , signers/2
+        , version/1
+        , serialization_template/1
+        , serialize/1
+        , deserialize/2
+        , for_client/1
+        , valid_at_protocol/2
+        ]).
+
+-export([
+          channel_id/1
+        , channel_pubkey/1
+        , role/1
+        , round/1
+        , state_hash/1
+        ]).
+
+-define(CHANNEL_CLIENT_RECONNECT_TX_TYPE, channel_client_reconnect_tx).
+-define(CHANNEL_CLIENT_RECONNECT_TX_FEE, 0).  % off-chain
+
+-define(INITIAL_VSN, 1).
+
+-type vsn() :: non_neg_integer().
+
+-record(channel_client_reconnect_tx,
+        {
+          channel_id   :: aeser_id:id()
+        , role         :: initiator | responder
+        , pub_key      :: aeser_id:id()
+        }).
+
+-opaque tx() :: #channel_client_reconnect_tx{}.
+
+-export_type([tx/0]).
+
+
+-spec new(map()) -> {ok, aetx:tx()}.
+new(#{ channel_id := ChannelId
+     , role       := Role
+     , pub_key    := Pubkey }) ->
+    channel = aeser_id:specialize_type(ChannelId),
+    account = aeser_id:specialize_type(Pubkey),
+    true = (Role == initiator) orelse (Role == responder),
```

Comment:
So you want to fail hard here if the role is unsupported? Which other values could be in `role`? And shouldn't that check be common for most of `aechannel`?

Reply:
I figured that it's in line with the assertions regarding channel_id and pubkey type on the lines above. `Role` is usually not part of the channel txs.